web-common/src/test/java/edu/unc/lib/boxc/web/common/services/AccessCopiesServiceTest.java

```diff
@@ -271,6 +272,33 @@ public void doesNotHavePlayableAudiofile() {
                 accessCopiesService.getDatastreamPid(mdObjectImg, principals, AUDIO_MIMETYPE_REGEX));
     }
 
+    @Test
+    public void primaryObjThumbnail() {
+        hasPermissions(mdObjectImg, true);
+        // Thumbnail object is the same as passed in object, so don't need to retrieve it again
+        assertNull(accessCopiesService.getThumbnailObject(mdObjectImg, principals));
+    }
+
+    @Test
+    public void noPrimaryObjThumbnailOneFile() {
+        hasPermissions(noOriginalFileObj, true);
+        hasPermissions(mdObjectImg, true);
+        List<ContentObjectSolrRecord> resultList = Collections.singletonList(mdObjectImg);
+        when(queryResponse.getBeans(ContentObjectSolrRecord.class)).thenReturn(resultList);
+        assertEquals(mdObjectImg, accessCopiesService.getThumbnailObject(noOriginalFileObj, principals));
+    }
+
+    @Test
+    public void noPrimaryObjThumbnailMultipleFiles() {
+        hasPermissions(noOriginalFileObj, true);
+        hasPermissions(mdObjectXml, true);
+        hasPermissions(mdObjectImg, true);
+        List<ContentObjectSolrRecord> resultList = Arrays.asList(mdObjectXml, mdObjectImg);
+        when(queryResponse.getBeans(ContentObjectSolrRecord.class)).thenReturn(resultList);
+        when(queryResponse.getResults().size()).thenReturn(resultList.size());
+        assertEquals(mdObjectImg, accessCopiesService.getThumbnailObject(noOriginalFileObj, principals));
+    }
+
```

Comment:
I feel like there should be a test for a work with files, but none with thumbnails. But if its all happening in a solr query then it might be hard to represent...